torch/csrc/jit/tensorexpr/codegen.cpp

```diff
@@ -83,21 +83,135 @@ void CodeGen::call_with_numel(void** args, int64_t numel) {
       false, "This codegen backend does not implement call_with_numel");
 }
 
-StmtPtr insertAllocFree(std::unordered_set<BufPtr>& interm_bufs, StmtPtr stmt) {
+c10::optional<size_t> bufSize(BufPtr buf) {
+  size_t size = elementSize(buf->dtype().scalar_type()) * buf->dtype().lanes();
+  for (auto& d : buf->dims()) {
+    if (!d->isConstant()) {
+      return c10::nullopt;
+    }
+    size = size * (*intValue(d));
+  }
+  return size;
+}
+
+// This algorithm takes the list of intermediate buffers and their liveness
+// ragnes, and returns a map of these buffers to memory blocks. A memory block
+// is indicated by the buffer for which it is allocated. Specifically, we
```

Comment:
What is a memory block? How different is it from Buf? If it's the same, can we avoid introducing a new concept?